scapy/arch/linux.py

```diff
@@ -161,6 +164,60 @@ def set_promisc(s,iff,val=1):
     s.setsockopt(SOL_PACKET, cmd, mreq)
 
 
+def get_alias_address(iface_name, ip_mask):
+    """
+    Get the correct source IP address of an interface alias
+    """
+
+    # Detect the architecture
+    bits, _ = platform.architecture()
+    if bits == "32bit":
+        offset, name_len = 32, 32
+    elif bits == "64bit":
+        offset, name_len = 16, 40
+    else:
+        warning("Unsupported architecture")
+        return []
+
+    # Retrieve interfaces structures
+    sck = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    names = array.array('B', '\0' * 4096)
+    ifreq = ioctl(sck.fileno(), SIOCGIFCONF,
+                  struct.pack("iL", len(names), names.buffer_info()[0]))
+
+    # Extract interfaces names
+    out = struct.unpack("iL", ifreq)[0]
+    names = names.tostring()
+    names = [names[i:i+offset].split('\0', 1)[0] for i in range(0, out, name_len)]
+
+    # Look for the IP address
+    ret_value = None
+    for ifname in names:
+        # Only look for a matching interface name
+        if not ifname.startswith(iface_name):
+            continue
+
+        # Retrieve and convert addresses
+        ifreq = ioctl(sck, SIOCGIFADDR, struct.pack("16s16x", ifname))
+        ifaddr = struct.unpack(">I", ifreq[20:24])[0]
+        ifreq = ioctl(sck, SIOCGIFNETMASK, struct.pack("16s16x", ifname))
+        msk = struct.unpack(">I", ifreq[20:24])[0]
+       
+        # Get the full interface name
+        if ':' in ifname:
+            ifname = ifname[:ifname.index(':')]
+        else:
+            continue
+
+        # Check if the source address is included in the network
+        if (ifaddr & msk) == ip_mask:
+            ret_value = (ifaddr & msk, msk, "0.0.0.0", ifname,
```

Comment:
You probably want to return that instead of affecting that variable and breaking. `ret_value` is useless since the function will return `None` if no `return` statement is reached.